File: 백준/Gold/1806. 부분합/부분합.py
import sys
lines = sys.stdin.read().strip().split("\n")
N, S = map(int, lines[0].split())
numbers = list(map(int, lines[1].split()))
    
left, right = 0, 0
sum = 0
min_length = sys.maxsize

while True:
    if sum >= S:
        min_length = min(min_length, right - left)
        sum -= numbers[left]
        left += 1
    elif right == N:
        break
    else:
        sum += numbers[right]
        right += 1

if min_length == sys.maxsize:
    print(0)
else:
    print(min_length)

File: 백준/Gold/1806. 부분합/부분합.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const [N, S] = input[0].split(' ').map(Number);

const numbers = input[1].split(' ').map(Number);

function minSubarrayLength(numbers, s) {
    let start = 0;
    let sum = 0;
    let minLen = Infinity;

    for (let end = 0; end < numbers.length; end++) {
        sum += numbers[end];

        while (sum >= s) {
            minLen = Math.min(minLen, end - start + 1);
            sum -= numbers[start];
            start += 1
        }
    }

    return minLen === Infinity ? 0 : minLen;
}

console.log(minSubarrayLength(numbers, S))